up_client_socket/rust/utransport-socket/src/lib.rs

```diff
@@ -0,0 +1,520 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+ mod constants;
+
+ use async_trait::async_trait;
+
+ use up_rust::UListener;
+ use up_rust::{UAttributesValidators, UriValidator};
+ use up_rust::{UCode, UMessage, UMessageType, UStatus, UTransport, UUri};
+ 
+ use crate::constants::BYTES_MSG_LENGTH;
+ use crate::constants::DISPATCHER_ADDR;
+ use protobuf::Message;
+ use std::collections::hash_map::Entry;
+ use std::collections::HashSet;
+ use std::io::{Read, Write};
+ use std::net::TcpStream;
+ use std::{
+     collections::HashMap,
+     sync::{Arc, Mutex},
+ };
+ use tokio::task;
+ use up_rust::ComparableListener;
+ use log::{error, debug};
+
+ pub struct UTransportSocket {
+     socket_sync: TcpStream,
+     listener_map: Arc<Mutex<HashMap<UUri, HashSet<ComparableListener>>>>,
+ }
+ 
+ impl UTransportSocket {
+     /// # Panics
+     ///
+     /// Will panic if issue connecting in sync socket
+     pub fn new() -> Result<Self, UStatus> {
+    
+        let socket_sync = if let Ok(socket) = TcpStream::connect(DISPATCHER_ADDR) {
+            socket
+        } else {
+            error!("Error connecting sync socket");
+            return Err(UStatus::fail_with_code(UCode::INTERNAL, "Issue in connecting sync socket"));
+        };
+    
+       let listener_map = Arc::new(Mutex::new(HashMap::new())) ;
+
+       let socket_clone = match socket_sync.try_clone() {
+        Ok(socket) => socket,
+        Err(err) => {
+      
+            error!("Issue in cloning sync socket: {}", err);
+            return Err(UStatus::fail_with_code(UCode::INTERNAL, "Issue in cloning socket_sync"));
+        }
+    };
+       let mut transport_socket = UTransportSocket {
+        socket_sync:socket_clone,
+        listener_map: listener_map.clone(),
+    };
+        match transport_socket.socket_init(){
```

Comment:
Since not using the Ok() value anyway consider this:
```rust
if let Err(err) = transport_socket.socket_init() {
 // should also refactor to have the error spelled out once time and re-use for the returned
 // Err as well 
 error!("Socket transport initialization failed: {}", err);
 return Err(UStatus::fail_with_code(UCode::INTERNAL, "Issue in initializing socket transport"));
}
debug!("Socket transport initialized successfully");
```